### PR TITLE
Add Arm functions

### PR DESCRIPTION
**Pull request by cameronfrandsen**
Add arm_init.c, filter_neon.S, and filter_neon_intrinsics.c to fix arm undefined references.

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -107,6 +107,8 @@ WXZLIB_OBJECTS =  \
 WXPNG_CFLAGS = -DNDEBUG $(__INC_ZLIB_p) $(____SHARED) $(CPPFLAGS) $(CFLAGS)
 WXPNG_OBJECTS =  \
 	wxpng_png.o \
+	wxpng_arm_init.o \
```
Comment (cameronfrandsen) [inline Makefile.in:110]: I needed to add the files here so that other libraries could use the functions (wxx). The PNG library does some other stuff with all three files if present and is able to:  https://github.com/smanders/wxWidgets/blob/xp3.1.0/src/png/Makefile.am#L82
This is fixed in later versions of wxWidgets. 
Later versions of wxWidgets have all three of these files (which is where I got them from). They do not add the files here though because wxWidgets starts using cmake. 
Basically what I am saying is these changes are not necessary once (or if) we update wxWidgets.

**Comment by smanders** [comment]
for future reference: it's helpful to know where these files came from... the commit or branch or something like that -- I'm spending the time now to figure this out, so we have a history and can compare (if necessary) -- this is especially important when you don't cherry-pick commits

**Comment by smanders** [comment]
the three new files
* src/png/arm/arm_init.c
* src/png/arm/filter_neon_intrinsics.c
* src/png/arm/filter_neon.S

match the versions from the v3.1.1 tag of wxWidgets (png is a submodule in this release of wxWidgets)

the Makefile changes here in this pull build the .c files, but it seems we do nothing with the .S file (for now)
